### ISSUE TITLE
Successful fast reload can be cancelled

### ISSUE THREAD

**Issue by BenLubar** [ReactiveDrop/reactivedrop_public_src#443]
Imported from <https://steamcommunity.com/app/563560/discussions/2/6670425060425619954/>:

> **\-Fast reload quick swap:**
> (this one has been there since the beginning of the game)
> When you succeed in a fast reload but changing weapons too quickly this is not taken into account and your weapon is not reloaded.

I've seen people doing this intentionally as a way to pass the time during missions when there aren't enemies. Sounds like another input validation bug.

---

This might be related to fast reload's prediction - the fast reload happens on the client but the server sees the weapon swap before the reload. Should also look into making the fast reload marker consistently placed for spectators, because that's probably the same root cause.

**Comment by jhh8** [comment]
Is that really a bug? I thought its a little delay to let the reload animation fully finish. Sometimes its even useful - ive had many cases where there was 20% of ammo on the floor, i would fast reload with a GL to refill it fully just to see someone take the ammo and i would cancel it by switching weapons

**Comment by GmaOrange** [comment]
Yes except just making it for fun above is a good example of real use.
